File: 414. Third Maximum Number/Python/Solution.py
class Solution(object):
    def thirdMax(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        first = -sys.maxsize
        second = first
        third = second
        for num in nums:
            if num > first:
                first, second, third = num, first, second
            elif num > second and num < first:
                second, third = num, second
            elif num > third and num < second:
                third = num
        if third == -sys.maxsize:
            return first
        return third

File: 414. Third Maximum Number/Java/Solution.java
public class Solution {
    public int thirdMax(int[] nums) {
        long first = Long.MIN_VALUE;
        long second = Long.MIN_VALUE;
        long third = Long.MIN_VALUE;
        for (int num : nums) {
            if (num > first) {
                third = second;
                second = first;
                first = num;
            } else if (num > second && num < first) {
                third = second;
                second = num;
            } else if (num >= third && num < second) {
                third = num;
            }
        }
        if (third == Long.MIN_VALUE) {
            return (int) first;
        }
        return (int) third;
    }
}